### PR TITLE
Switch to Bevy 0.2

### LINKED ISSUE #12: Bevy 0.2
We should update to Bevy 0.2.

### PR DESCRIPTION
**Pull request by ocornoc**
There's a problem with the `cam_contmove_parent_test` example, as I haven't finished the logic for parented transformations.

Resolves #12

### CODE REVIEW

File: crates/bounded_planet_core/src/camera/mod.rs [hunk 1]
```diff
@@ -197,64 +197,108 @@ impl UniversalGeometry {
...
+        opt: Transform,
+        oct: Transform,
+        s: Vec3,
+        scale: f32
+    ) -> (Transform, Transform) {
+        match self.0 {
+            UniversalGeometry::Plane {..} => {
+                let mut out = self.noparent_trans(Transform::new(*oct.value()), s, scale);
+                let outi = out.value_mut();
+                // insert appropriate logic here
```
Comment (martindevans) [inline crates/bounded_planet_core/src/camera/mod.rs]: use `//todo(#issue_number)` here

File: crates/bounded_planet_core/src/camera/mod.rs [hunk 2]
```diff
@@ -197,64 +197,108 @@ impl UniversalGeometry {
...
+        opt: Transform,
+        oct: Transform,
+        s: f32,
+        scale: f32
+    ) -> (Transform, Transform) {
+        match self.0 {
+            UniversalGeometry::Plane {..} => {
+                let mut out = self.noparent_zoom(Transform::new(*oct.value()), s, scale);
+                let outi = out.value_mut();
+                // insert appropriate logic here
```
Comment (martindevans) [inline crates/bounded_planet_core/src/camera/mod.rs]: use `//todo(#issue_number)` here

**Review by martindevans — CHANGES_REQUESTED** [review]

File: crates/bounded_planet_core/src/camera/mod.rs [hunk 3]
```diff
@@ -166,13 +166,13 @@ impl CameraBPAction {
 /// Origin fields are expected to be wrt. the origin used by [`CameraBP`]s.
 #[derive(Copy, Clone)]
```
Comment (Jerald) [inline crates/bounded_planet_core/src/camera/mod.rs]:
```suggestion
#[derive(Debug, Copy, Clone)]
```

File: crates/bounded_planet_core/src/camera/mod.rs [hunk 4]
```diff
@@ -197,64 +197,108 @@ impl UniversalGeometry {
...
-    /// Get the new position and rotation resulting from the original position
-    /// `p`, original rotation `r`, and movement `s` about `self` (relative to
-    /// `r`).
-    fn trans(&self, p: Vec3, r: Quat, s: Vec3, scale: f32) -> (Vec3, Quat) {
-        fn plane(_o: Vec3, n: Vec3, p: Vec3, r: Quat, s: Vec3, scale: f32) -> (Vec3, Quat) {
-            let mut delta = r.mul_vec3(s);
+/// A private newtype of Universal Geometry, that satisfies some invariants:
+/// 1) `self.0` is normalized.
+/// 2) If `self.0` is a plane, then its normal is a unit vector.
+#[derive(Copy, Clone)]
```
Comment (Jerald) [inline crates/bounded_planet_core/src/camera/mod.rs]:
```suggestion
#[derive(Debug, Copy, Clone)]
```

**Review by Jerald — DISMISSED** [review]

File: crates/bounded_planet_core/src/camera/mod.rs (hunk 1 above)
Comment (ocornoc) [reply crates/bounded_planet_core/src/camera/mod.rs]: See the description, this is the logic I haven't finished.

File: crates/bounded_planet_core/src/camera/mod.rs (hunk 2 above)
Comment (ocornoc) [reply crates/bounded_planet_core/src/camera/mod.rs]: See the description, this is the logic I haven't finished.

**Comment by ocornoc** [comment]
Adds a regression to parented cameras, my matrix math isn't correct.

**Review by martindevans — APPROVED** [review]

**Review by Avril112113 — APPROVED** [review]

**Review by Nyefari — APPROVED** [review]

**Pull request merged**